cartographer/io/image.cc

```diff
@@ -9,6 +9,21 @@ namespace cartographer {
 namespace io {
 namespace {
 
+uint32_t ColorToCairo(const Color& color) {
+  uint32_t r = ColorComponentToUint8(color[0]);
+  uint32_t g = ColorComponentToUint8(color[1]);
+  uint32_t b = ColorComponentToUint8(color[2]);
+  return (255 << 24) | (r << 16) | (g << 8) | b;
```

Comment:
The (255 << 24) is undefined behavior: Cast the 255 to `uint32` before shifting?